### PR TITLE
added changes for data augmentation

### PR DESCRIPTION
**Pull request by ykarmesh**
## Motivation and Context

Adds data augmentation capabilities to Object Rearrange Codebase

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: habitat-baselines/habitat_baselines/rl/ddppo/transforms.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        shift = torch.randint(
+            0, 2 * self.pad + 1, size=(n, 1, 1, 2), device=x.device, dtype=x.dtype
+        )
+        shift *= 2.0 / (h + 2 * self.pad)
+
+        grid = base_grid + shift
+        return F.grid_sample(x, grid, padding_mode="zeros", align_corners=False)
+
+
+class Transform:
```
Comment (cpaxton) [inline habitat-baselines/habitat_baselines/rl/ddppo/transforms.py:37]: is this supposed to be an abstract base class? maybe a comment or include ABC here?

File: habitat-baselines/habitat_baselines/rl/ddppo/transforms.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision.transforms.functional as TF
+from torchvision.transforms import ColorJitter, RandomApply
+
+
+class RandomShiftsAug(nn.Module):
```
Comment (cpaxton) [inline habitat-baselines/habitat_baselines/rl/ddppo/transforms.py:10]: class description - when should i use this? can be simple

File: habitat-baselines/habitat_baselines/rl/ddppo/transforms.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        x = torch.cat([self.apply(imgs) for imgs in x])
+
+        # put timestep (t) first
+        _, A, B, C = x.shape
+        x = torch.einsum("ntabc->tnabc", x.view(N, T, A, B, C)).flatten(0, 1)
+
+        return x
+
+
+class ResizeTransform(Transform):
```
Comment (cpaxton) [inline habitat-baselines/habitat_baselines/rl/ddppo/transforms.py:74]: see above

File: habitat-baselines/habitat_baselines/rl/ddppo/transforms.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        x = x.float() / 255.0
+        return x
+
+
+class ShiftAndJitterTransform(Transform):
+    def __init__(self, size):
+        self.size = size
+
+    def apply(self, x):
+        # x = x.permute(0, 3, 1, 2)
```
Comment (cpaxton) [inline habitat-baselines/habitat_baselines/rl/ddppo/transforms.py:91]: dead code?

**Review by cpaxton — APPROVED** [review]
better documentation but mostly lgtm

File: habitat-baselines/habitat_baselines/rl/ddppo/transforms.py (hunk 1 above)
Comment (ykarmesh) [reply habitat-baselines/habitat_baselines/rl/ddppo/transforms.py]: changed

File: habitat-baselines/habitat_baselines/rl/ddppo/transforms.py (hunk 2 above)
Comment (ykarmesh) [reply habitat-baselines/habitat_baselines/rl/ddppo/transforms.py:10]: done

File: habitat-baselines/habitat_baselines/config/default_structured_configs.py [hunk 5]
```diff
@@ -237,6 +237,7 @@ class PolicyConfig(HabitatBaselinesBaseConfig):
     hierarchical_policy: HierarchicalPolicy = MISSING
     ovrl: bool = False
     no_downscaling: bool = False
+    use_augmentations: bool = False
```
Comment (yvsriram) [inline habitat-baselines/habitat_baselines/config/default_structured_configs.py:240]: Should we enable them by default in our configs?

**Review by yvsriram — APPROVED** [review]
LGTM

File: habitat-baselines/habitat_baselines/config/default_structured_configs.py (hunk 5 above)
Comment (ykarmesh) [reply habitat-baselines/habitat_baselines/config/default_structured_configs.py:240]: Lets do it after we have tested it once on the new dataset

**Pull request merged**